// material-ui-pickers-v4/_shared/ModalDialog.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.styles = exports.ModalDialog = void 0;
var React = __importStar(require("react"));
var clsx_1 = __importDefault(require("clsx"));
var core_1 = require("@material-ui/core");
var dimensions_1 = require("../constants/dimensions");
exports.ModalDialog = function (_a) {
    var _b, _c;
    var children = _a.children, classes = _a.classes, onAccept = _a.onAccept, onDismiss = _a.onDismiss, onClear = _a.onClear, onSetToday = _a.onSetToday, okLabel = _a.okLabel, cancelLabel = _a.cancelLabel, clearLabel = _a.clearLabel, todayLabel = _a.todayLabel, clearable = _a.clearable, showTodayButton = _a.showTodayButton, showTabs = _a.showTabs, wider = _a.wider, DialogComponent = _a.DialogComponent, DialogActionsComponent = _a.DialogActionsComponent, DialogContentComponent = _a.DialogContentComponent, other = __rest(_a, ["children", "classes", "onAccept", "onDismiss", "onClear", "onSetToday", "okLabel", "cancelLabel", "clearLabel", "todayLabel", "clearable", "showTodayButton", "showTabs", "wider", "DialogComponent", "DialogActionsComponent", "DialogContentComponent"]);
    var DefaultDialog = DialogComponent || core_1.Dialog;
    var DefaultDialogContent = DialogContentComponent || core_1.DialogContent;
    var DefaultDialogActions = DialogActionsComponent || core_1.DialogActions;
    return (React.createElement(DefaultDialog, __assign({ role: "dialog", onClose: onDismiss, classes: {
            paper: clsx_1.default(classes.dialogRoot, (_b = {},
                _b[classes.dialogRootWider] = wider,
                _b)),
        } }, other),
        React.createElement(DefaultDialogContent, { children: children, className: classes.dialog }),
        React.createElement(DefaultDialogActions, { classes: {
                root: clsx_1.default((_c = {},
                    _c[classes.withAdditionalAction] = clearable || showTodayButton,
                    _c)),
            } },
            clearable && (React.createElement(core_1.Button, { color: "primary", onClick: onClear }, clearLabel)),
            showTodayButton && (React.createElement(core_1.Button, { color: "primary", onClick: onSetToday }, todayLabel)),
            cancelLabel && (React.createElement(core_1.Button, { color: "primary", variant: "outlined", onClick: onDismiss }, cancelLabel)),
            okLabel && (React.createElement(core_1.Button, { color: "primary", variant: "contained", onClick: onAccept }, okLabel)))));
};
exports.ModalDialog.displayName = 'ModalDialog';
exports.styles = core_1.createStyles({
    dialogRoot: {
        minWidth: dimensions_1.DIALOG_WIDTH,
        maxWidth: dimensions_1.DIALOG_WIDTH,
    },
    dialogRootWider: {
        minWidth: dimensions_1.DIALOG_WIDTH_WIDER,
    },
    dialog: {
        '&:first-child': {
            padding: 0,
        },
    },
    withAdditionalAction: {
        // set justifyContent to default value to fix IE11 layout bug
        // see https://github.com/dmtrKovalenko/material-ui-pickers/pull/267
        justifyContent: 'flex-start',
        '& > *:first-child': {
            marginRight: 'auto',
        },
    },
});
exports.default = core_1.withStyles(exports.styles, { name: 'MuiPickersModal' })(exports.ModalDialog);
