// material-ui-pickers-v4/wrappers/StaticWrapper.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticWrapper = void 0;
var React = __importStar(require("react"));
var core_1 = require("@material-ui/core");
var dimensions_1 = require("../constants/dimensions");
var useStyles = core_1.makeStyles(function (theme) { return ({
    staticWrapperRoot: {
        overflow: 'hidden',
        minWidth: dimensions_1.DIALOG_WIDTH,
        display: 'flex',
        flexDirection: 'column',
        backgroundColor: theme.palette.background.paper,
    },
}); }, { name: 'MuiPickersStaticWrapper' });
exports.StaticWrapper = function (_a) {
    var children = _a.children;
    var classes = useStyles();
    return React.createElement("div", { className: classes.staticWrapperRoot, children: children });
};
