// material-ui-pickers-v4/views/Year/YearView.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.YearSelection = exports.useStyles = void 0;
var React = __importStar(require("react"));
var Year_1 = __importDefault(require("./Year"));
var core_1 = require("@material-ui/core");
var useUtils_1 = require("../../_shared/hooks/useUtils");
var Wrapper_1 = require("../../wrappers/Wrapper");
exports.useStyles = core_1.makeStyles({
    container: {
        height: 300,
        overflowY: 'auto',
    },
}, { name: 'MuiPickersYearSelection' });
exports.YearSelection = function (_a) {
    var date = _a.date, onChange = _a.onChange, onYearChange = _a.onYearChange, minDate = _a.minDate, maxDate = _a.maxDate, disablePast = _a.disablePast, disableFuture = _a.disableFuture, animateYearScrolling = _a.animateYearScrolling;
    var utils = useUtils_1.useUtils();
    var classes = exports.useStyles();
    var currentVariant = React.useContext(Wrapper_1.VariantContext);
    var selectedYearRef = React.useRef(null);
    React.useEffect(function () {
        if (selectedYearRef.current && selectedYearRef.current.scrollIntoView) {
            try {
                selectedYearRef.current.scrollIntoView({
                    block: currentVariant === 'static' ? 'nearest' : 'center',
                    behavior: animateYearScrolling ? 'smooth' : 'auto',
                });
            }
            catch (e) {
                // call without arguments in case when scrollIntoView works improperly (e.g. Firefox 52-57)
                selectedYearRef.current.scrollIntoView();
            }
        }
    }, []); // eslint-disable-line
    var currentYear = utils.getYear(date);
    var onYearSelect = React.useCallback(function (year) {
        var newDate = utils.setYear(date, year);
        if (onYearChange) {
            onYearChange(newDate);
        }
        onChange(newDate, true);
    }, [date, onChange, onYearChange, utils]);
    return (React.createElement("div", { className: classes.container }, utils.getYearRange(minDate, maxDate).map(function (year) {
        var yearNumber = utils.getYear(year);
        var selected = yearNumber === currentYear;
        return (React.createElement(Year_1.default, { key: utils.getYearText(year), selected: selected, value: yearNumber, onSelect: onYearSelect, ref: selected ? selectedYearRef : undefined, disabled: Boolean((disablePast && utils.isBeforeYear(year, utils.date())) ||
                (disableFuture && utils.isAfterYear(year, utils.date()))) }, utils.getYearText(year)));
    })));
};
